File: C++/loops.cpp
//loops in c++:
#include <iostream>
using namespace std;

int main() {
    //for loop:
    cout<<"For loop:"<<endl;
    for(int i=0; i<5; i++){
        cout<<i<<" ";
    }
    cout<<endl;

    //while loop:
    cout<<"While loop:"<<endl;
    int j=0;
    while(j<5){
        cout<<j<<" ";
        j++;
    }
    cout<<endl;

    //do-while loop:
    cout<<"Do-while loop:"<<endl;
    int k=0;
    do{
        cout<<k<<" ";
        k++;
    }while(k<5);
    cout<<endl;

    return 0;
}

File: Rust/programs/src/loops.rs
//loops in rust:
pub fn loop() {
    //for loop
    let arr = [10, 20, 30, 40, 50]; 
    for i in 0..arr.len() {
        println!("Element at index {}: {}", i, arr[i]);
    }
    //while loop
    let mut count = 0;
    while count < 5 {
        println!("Count: {}", count);
        count += 1;
    }
    //loop (infinite loop)
    let mut num = 0;
    loop {
        if num >= 5 {
            break; // exit the loop when num is 5 or greater
        }
        println!("Number: {}", num);
        num += 1;
    }
    //for-each loop
    let vec = vec![1, 2, 3, 4, 5];
    for element in &vec {
        println!("Element: {}", element);
    }
}